wave_lang/kernel/wave/generate_bounds_exprs.py

```diff
@@ -52,20 +64,22 @@ def is_read_write(node: fx.Node):
 
         vector_shapes = node.vector_shapes or hardware_constraint.vector_shapes
         is_shared_mem = is_shared_mem_access(node)
-        bounds = find_index_bounds(
+        bounds, unconstrained_dim = find_index_bounds(
             constraints, node.index, vector_shapes, node.type.symbolic_shape
         )
         if is_shared_mem and bounds:
             bounds = remove_global_indexing(bounds, constraints)
             # Masking against global bounds was already handled when reading from
             # global mem, but we still need to handle masking against vector
             # size during shared mem access.
-            # A bound expression for this case is `min(global_bound, vector_size)`.
-            # Replace global bound with `max(tile_size, vector_size)` so the entire
-            # expression `min(max(tile_size, vector_size), vector_size)` can be
-            # simplified to just vector size.
             bounds = {
-                k: safe_subs(v, {k: _get_max_tile_size(k, constraints, vector_shapes)})
+                k: (
+                    _get_unconstrained_tile_size(node)
```

Comment:
I think we can just always replace `_get_max_tile_size` with `_get_unconstrained_tile_size` and don't need `unconstrained_dim`.

Reply:
Yeah that fails the test too

Reply:
Something like
```
            bounds = {
                k: safe_subs(v, {k: get_custom(node.memory).unpadded_shape[k]})
                for k, v in bounds.items()
            }
```

Reply:
I'm still not certain that constrained dimension needs to have unpadded shape as the bound